store/views.py

```diff
@@ -1,82 +1,263 @@
+import datetime
+from django.http import Http404
 from django.shortcuts import render
 from django.shortcuts import get_object_or_404
 from store.models import *
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
 
-# Create your views here.
 
 def index(request):
     return render(request, 'store/index.html')
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
-    context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
-    }
-    # START YOUR CODE HERE
     
+    # List of available book id(s)
+    avail = []
+    for i in Book.objects.all():
+        avail.append(i.id)
     
+    # If the id in the request is in the list,  
+    if bid in avail:
+        book = Book.objects.filter(id = bid).first()
+        available_copies = BookCopy.objects.filter(book = book).filter(status = True).count()
+    # Else return page not found
+    else:
+        raise Http404('Book does not exist')
+
+    ## USER GIVEN RATING OF THE BOOK ##
+
+    # All rating objects for the book
+    totalRatings = UserRating.objects.filter(book = book)
+    # check if user is logged in
+    # and rating objects are present for the book
+    if request.user.is_authenticated and totalRatings.count() > 0:
+        myRating = totalRatings.filter(user_id = request.user.id).first()
+        # If user rating is found
+        if myRating != None:
+            givenRating = myRating.rating
+    # Else the rating for the book by the user is None
+    else:
+        givenRating = None
+
+
+    context = {
+        'book': book,
+        'num_available': available_copies,
+        'usrating': givenRating,
```

Comment:
Consider the case when another user has rated the book, and the current user hasn't done so. Then the local variable `givenRating` won't be assigned any value, which will raise an error.